plume-front/src/editor.rs

```diff
@@ -0,0 +1,208 @@
+
+            let content = placeholder(make_editable("article").try_into().unwrap(), &i18n!(CATALOG, "Write your article here. Markdown is supported."));
+            if !content_val.is_empty() {
+                content.dataset().insert("edited", "true").unwrap();
+            }
+            content.append_child(&document().create_text_node(&content_val));
+
+            // Add the elements, and make sure the placeholders are rendered
+            ed.append_child(&title);
+            title.focus();
+            title.blur();
+            ed.append_child(&subtitle);
+            subtitle.focus();
+            subtitle.blur();
+            ed.append_child(&content);
+            content.focus();
+            content.blur();
+
+            let widgets = Rc::new((
+                title, subtitle, content
+            ));
+            document().get_element_by_id("publish")?.add_event_listener(mv!(old_ed, widgets => move |_: ClickEvent| {
+                document().get_element_by_id("publish-popup")
+                    .unwrap_or_else(mv!(old_ed, widgets => move || {
+                        let popup = document().create_element("div").unwrap();
+                        popup.class_list().add("popup").unwrap();
+                        popup.set_attribute("id", "publish-popup").unwrap();
+
+                        let tags = get_elt_value("tags").split(',').map(str::trim).map(str::to_string).collect::<Vec<_>>();
+                        let license = get_elt_value("license");
+                        make_input(&i18n!(CATALOG, "Tags"), "popup-tags", &popup).set_raw_value(&tags.join(", "));
+                        make_input(&i18n!(CATALOG, "License"), "popup-license", &popup).set_raw_value(&license);
+
+                        let cover_label = document().create_element("label").unwrap();
+                        cover_label.append_child(&document().create_text_node(&i18n!(CATALOG, "Cover")));
+                        cover_label.set_attribute("for", "cover").unwrap();
+                        let cover = document().get_element_by_id("cover").unwrap();
+                        cover.parent_element().unwrap().remove_child(&cover).ok();
+                        popup.append_child(&cover_label);
+                        popup.append_child(&cover);
+
+                        let button = document().create_element("input").unwrap();
+                        js!{
+                            @{&button}.type = "submit";
+                            @{&button}.value = @{i18n!(CATALOG, "Publish")};
+                        };
+                        button.append_child(&document().create_text_node(&i18n!(CATALOG, "Publish")));
+                        button.add_event_listener(mv!(widgets, old_ed => move |_: ClickEvent| {
+                            set_value("title", widgets.0.inner_text());
+                            set_value("subtitle", widgets.1.inner_text());
+                            set_value("editor-content", widgets.2.inner_text());
```

Comment:
By doing the following change, I was able to copy past a picture in a post and have it rendered as expected. However when re-editing the article, `<` and `>` where escaped, leading to a messy html tag in the middle of the content. I don't know if this is considered in scope for this pr, but this might be something to look at, as it will get important once the editor get richer

```suggestion
                            set_value("editor-content", js!{return @{&widgets.2}.innerHTML}.as_str().unwrap_or_default());
```

Reply:
I think it would be easy to avoid escaping here, I will try.